cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java

```diff
@@ -389,17 +390,12 @@ public OutputStream getOutput() throws IOException {
   }
 
   protected String toHBaseNamespace(Id.Namespace namespace) {
-    return "cdap_" + namespace.getId();
+    // Handle backward compatibility to not add the prefix for default namespace
+    // TODO: Conditional should be removed when we have a way to upgrade user datasets
```

Comment:
The decision we made was that we're not going to upgrade user datasets in 2.8.0. Only system datasets will be upgraded. In a later release, when we are able to upgrade user datasets, we will remove this check.